### PR TITLE
search: disable structural search by default

### PR DESCRIPTION
**Pull request by stefanhengl**
This disables structural search by default. As a consequence the toggle for structural search `[ ]` will be hidden in the UI and API calls with a structural search query will fail.

Users can re-enable structural search by setting `experimentalFeatures.structuralSearch: "enabled"` in site configuration.

Note: The feature is not deprecated. We are merely switching the default from "enabled" to "disabled".

Test plan:
updated unit tests: a couple of tests expected structural search to be on by default. I updated the tests to enable structural search explicitly.

### CODE REVIEW

**Review by mrnugget — APPROVED** [review]

**Comment by keegancsmith** [comment]
We should get approval from others before this ends up in a release. Additionally we need a plan for environments we do continuous deployments. Even though this is just disabling by default, given there is product buy in for this feature (as low maintained as it is), this can be seen as a big change.

File: dev/gqltest/search_test.go [hunk 1]
```diff
@@ -723,6 +719,31 @@ func testSearchClient(t *testing.T, client searchClient) {
...
+			if err != nil {
+				t.Fatal(err)
+			}
+		}()
+
+		siteConfig.ExperimentalFeatures = &schema.ExperimentalFeatures{StructuralSearch: "enabled"}
+		err = client.UpdateSiteConfiguration(siteConfig, lastID)
+		if err != nil {
+			t.Fatal(err)
+		}
```
[the remark this answers is not in the document]
Comment (unknwon) [reply dev/gqltest/search_test.go:745]: The client and the Sourcegraph server isn't sharing the memory (two processes)?

[the remark this answers is not in the document]
Comment (mrnugget) [reply dev/gqltest/search_test.go:745]: Ah! Right, we're in in `gqltest`!

**Review by jtibshirani — APPROVED** [review]
Looks good to me 🎉

**Review by keegancsmith — APPROVED** [review]
Funny that the feature toggle is still under experimental features...

Ship it!

**Pull request merged**